[Travis CI] build failed soci_mysql on XCode clang

https://travis-ci.com/tt4g/conan-soci/jobs/217773644#L6057

```bash
make[2]: *** No rule to make target `/Users/travis/.conan/data/mysql-connector-c/6.1.11/bincrafters/stable/package/a65ed1b6fe2e3ea0a48cea3173752e57dcb29d19/lib/libmysqlclient.so', needed by `lib/libsoci_mysql.4.0.0.dylib'.  Stop.
make[2]: *** Waiting for unfinished jobs....
[ 83%] Building CXX object source_subfolder/src/backends/mysql/CMakeFiles/soci_mysql.dir/vector-use-type.cpp.o
[ 85%] Building CXX object source_subfolder/src/backends/postgresql/CMakeFiles/soci_postgresql.dir/row-id.cpp.o
[ 87%] Building CXX object source_subfolder/src/backends/postgresql/CMakeFiles/soci_postgresql.dir/session.cpp.o
make[1]: *** [source_subfolder/src/backends/mysql/CMakeFiles/soci_mysql.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 88%] Building CXX object source_subfolder/src/backends/postgresql/CMakeFiles/soci_postgresql.dir/standard-into-type.cpp.o
[ 90%] Building CXX object source_subfolder/src/backends/postgresql/CMakeFiles/soci_postgresql.dir/standard-use-type.cpp.o
[ 92%] Building CXX object source_subfolder/src/backends/postgresql/CMakeFiles/soci_postgresql.dir/statement.cpp.o
[ 94%] Building CXX object source_subfolder/src/backends/postgresql/CMakeFiles/soci_postgresql.dir/vector-into-type.cpp.o
[ 96%] Building CXX object source_subfolder/src/backends/postgresql/CMakeFiles/soci_postgresql.dir/vector-use-type.cpp.o
[ 98%] Linking CXX shared library ../../../../lib/libsoci_postgresql.dylib
[ 98%] Built target soci_postgresql
make: *** [all] Error 2
SOCI/4.0.0@user/channel: 
SOCI/4.0.0@user/channel: ERROR: Package 'adeec79edcd226d11d819765a3ec5e8dc6fe227d' build failed
```

--- a/conanfile.py
+++ b/conanfile.py
@@ -22,40 +22,15 @@ class SociConan(ConanFile):
         "soci_tests": [True, False],
         "with_boost": [True, False],
         "soci_empty": [True, False],
-        "soci_empty_test_connstr": "ANY",
         "with_db2": [True, False],
-        "db2_include_dir": "ANY",
-        "db2_libraries": "ANY",
-        "soci_db2_test_connstr": "ANY",
         "with_firebird": [True, False],
-        "firebird_include_dir": "ANY",
-        "firebird_libraries": "ANY",
-        "soci_firebird_test_connstr": "ANY",
         "soci_firebird_embedded": [True, False],
         "with_mysql": [True, False],
-        "mysql_dir": "ANY",
-        "mysql_include_dir": "ANY",
-        "mysql_libraries": "ANY",
-        "soci_mysql_test_connstr": "ANY",
         "with_odbc": [True, False],
-        "odbc_include_dir": "ANY",
-        "odbc_libraries": "ANY",
-        "soci_odbc_test_access_connstr": "ANY",
-        "soci_odbc_test_mysql_connstr": "ANY",
-        "soci_odbc_test_postgresql_connstr": "ANY",
         "with_oracle": [True, False],
-        "oracle_include_dir": "ANY",
-        "oracle_libraries": "ANY",
-        "soci_oracle_test_connstr": "ANY",
         "with_postgresql": [True, False],
-        "postgresql_include_dir": "ANY",
-        "postgresql_libraries": "ANY",
-        "soci_postgresql_test_connstr": "ANY",
         "soci_postgresql_nosinglerowmode": [True, False],
-        "with_sqlite3": [True, False],
-        "sqlite_include_dir": "ANY",
-        "sqlite_libraries": "ANY",
-        "soci_sqlite_test_connstr": "ANY"
+        "with_sqlite3": [True, False]
     }
     default_options = {
         "shared": True,
@@ -63,136 +38,22 @@ class SociConan(ConanFile):
         "soci_tests": False,
         "with_boost": True,
         "soci_empty": True,
-        "soci_empty_test_connstr": None,
         "with_db2": False,
-        "db2_include_dir": None,
-        "db2_libraries": None,
-        "soci_db2_test_connstr": None,
         "with_firebird": False,
-        "firebird_include_dir": None,
-        "firebird_libraries": None,
-        "soci_firebird_test_connstr": None,
         "soci_firebird_embedded": False,
         "with_mysql": False,
-        "mysql_dir": None,
-        "mysql_include_dir": None,
-        "mysql_libraries": None,
-        "soci_mysql_test_connstr": None,
         "with_odbc": False,
-        "odbc_include_dir": None,
-        "odbc_libraries": None,
-        "soci_odbc_test_access_connstr": None,
-        "soci_odbc_test_mysql_connstr": None,
-        "soci_odbc_test_postgresql_connstr": None,
         "with_oracle": False,
-        "oracle_include_dir": None,
-        "oracle_libraries": None,
-        "soci_oracle_test_connstr": None,
         "with_postgresql": False,
-        "postgresql_include_dir": None,
-        "postgresql_libraries": None,
-        "soci_postgresql_test_connstr": None,
         "soci_postgresql_nosinglerowmode": False,
-        "with_sqlite3": False,
-        "sqlite_include_dir": None,
-        "sqlite_libraries": None,
-        "soci_sqlite_test_connstr": None
+        "with_sqlite3": False
     }
     generators = "cmake"
     _source_subfolder = "source_subfolder"
     _build_subfolder = "build_subfolder"
 
-    def _requires_db2(self):
-        """check requires conan db2.
-
-        :returns: true if require conan db2, otherwise false.
-        :rtype: bool
-        """
-        # use db2 when not defined db2_include_dir and db2_libraries
-        return (not self.options.db2_include_dir
-                and not self.options.db2_libraries)
-
-    def _requires_firebird(self):
-        """check requires conan firebird.
-
-        :returns: true if require conan firebird, otherwise false.
-        :rtype: bool
-        """
-        # use firebird when not defined firebird_include_dir and firebird_libraries
-        return (not self.options.firebird_include_dir
-                and not self.options.firebird_libraries)
-
-    def _requires_mysql(self):
-        """check requires conan mysql.
-
-        :returns: true if require conan mysql, otherwise false.
-        :rtype: bool
-        """
-        # use mysql-connector-c package if _set_mysql_paths() is true.
-        return self._set_mysql_paths()
-
-    def _set_mysql_paths(self):
-        """check CMake definition related to  MySQL backend path.
-
-        :returns: true if set mysql libraries path, otherwise false.
-        :rtype: bool
-        """
-        # if consumer not defined.
-        return (not self.options.mysql_dir
-                or (not self.options.mysql_include_dir
-                    and not self.options.mysql_libraries))
-
-    def _requires_odbc(self):
-        """check requires conan odbc.
-
-        :returns: true if require conan odbc, otherwise false.
-        :rtype: bool
-        """
-        # use odbc when not defined odbc_include_dir and odbc_libraries
-        return (not self.options.odbc_include_dir
-                and not self.options.odbc_libraries)
-
-    def _requires_oracle(self):
-        """check requires conan oracle.
-
-        :returns: true if require conan oracle, otherwise false.
-        :rtype: bool
-        """
-        # use oracle when not defined oracle_include_dir and oracle_libraries
-        return (not self.options.oracle_include_dir
-                and not self.options.oracle_libraries)
-
-    def _requires_libpq(self):
-        """check requires conan libpq.
-
-        :returns: true if require conan libpq, otherwise false.
-        :rtype: bool
-        """
-        # use libpq package if _set_postgresql_paths() is true.
-        return self._set_postgresql_paths()
-
-    def _set_postgresql_paths(self):
-        """check CMake definition related to  PostgreSQL backend path.
-
-        :returns: true if set libpq libraries path, otherwise false.
-        :rtype: bool
-        """
-        # if consumer not defined.
-        return (not self.options.postgresql_include_dir
-                and not self.options.postgresql_libraries)
-
-    def _requires_sqlite3(self):
-        """check requires conan sqlite3.
-
-        :returns: true if require conan sqlite3, otherwise false.
-        :rtype: bool
-        """
-        # use sqlite3 when not defined sqlite_include_dir and sqlite_libraries
-        return (not self.options.sqlite_include_dir
-                and not self.options.sqlite_libraries)
-
     def configure(self):
-        if (self._requires_mysql() and self.options.shared
+        if (self.options.with_mysql and self.options.shared
             and self.settings.os != "Windows"):
             # If use mysql-connector-c and SOCI:shared=True on UNIX-like OS,
             # link failed by ld command: "relocation R_X86_64_PC32 against symbol
@@ -206,35 +67,28 @@ class SociConan(ConanFile):
             self.requires("boost/1.70.0@conan/stable")
 
         if self.options.with_db2:
-            if self._requires_db2():
-                # not found conan db2 library.
-                pass
+            # conan db2 library not found.
+            pass
 
         if self.options.with_firebird:
-            if self._requires_firebird():
-                # not found conan firebird library.
-                pass
+            # conan firebird library not found.
+            pass
 
         if self.options.with_mysql:
-            if self._requires_mysql():
-                self.requires("mysql-connector-c/6.1.11@bincrafters/stable")
+            self.requires("mysql-connector-c/6.1.11@bincrafters/stable")
 
         if self.options.with_odbc:
-            if self._requires_odbc():
-                self.requires("odbc/2.3.7@bincrafters/stable")
+            self.requires("odbc/2.3.7@bincrafters/stable")
 
         if self.options.with_oracle:
-            if self._requires_oracle():
-                # not found conan oracle library.
-                pass
+            # conan oracle library not found.
+            pass
 
         if self.options.with_postgresql:
-            if self._requires_libpq():
-                self.requires("libpq/9.6.9@bincrafters/stable")
+            self.requires("libpq/9.6.9@bincrafters/stable")
 
         if self.options.with_sqlite3:
-            if self._requires_sqlite3():
-                self.requires("sqlite3/3.25.3@bincrafters/stable")
+            self.requires("sqlite3/3.25.3@bincrafters/stable")
 
     def source(self):
         # TODO: get source from archive after SOCI 4.0.0 released.
@@ -258,169 +112,18 @@ class SociConan(ConanFile):
         cmake.definitions["SOCI_CXX_C11"] = self.options.soci_cxx_c11
         cmake.definitions["SOCI_TESTS"] = self.options.soci_tests
         cmake.definitions["WITH_BOOST"] = self.options.with_boost
-
         cmake.definitions["SOCI_EMPTY"] = self.options.soci_empty
-        if self.options.soci_empty:
-            if self.options.soci_empty_test_connstr:
-                cmake.definitions["SOCI_EMPTY_TEST_CONNSTR"] = \
-                    self.options.soci_empty_test_connstr
-
         cmake.definitions["WITH_DB2"] = self.options.with_db2
-        if self.options.with_db2:
-            if not self._requires_db2():
-                cmake.definitions["DB2_INCLUDE_DIR"] = \
-                    self.options.db2_include_dir
-                cmake.definitions["DB2_LIBRARIES"] = self.options.db2_libraries
-
-            if self.options.soci_db2_test_connstr:
-                cmake.definitions["SOCI_DB2_TEST_CONNSTR"] = \
-                    self.options.soci_db2_test_connstr
-
         cmake.definitions["WITH_FIREBIRD"] = self.options.with_firebird
-        if self.options.with_firebird:
-            if self._requires_firebird():
-                cmake.definitions["FIREBIRD_INCLUDE_DIR"] = \
-                    self.options.firebird_include_dir
-                cmake.definitions["FIREBIRD_LIBRARIES"] = \
-                    self.options.firebird_libraries
-
-            if self.options.soci_firebird_test_connstr:
-                cmake.definitions["SOCI_FIREBIRD_TEST_CONNSTR"] = \
-                    self.options.soci_firebird_test_connstr
-
-            cmake.definitions["SOCI_FIREBIRD_EMBEDDED"] = \
+        cmake.definitions["SOCI_FIREBIRD_EMBEDDED"] = \
                 self.options.soci_firebird_embedded
-
         cmake.definitions["WITH_MYSQL"] = self.options.with_mysql
-        if self.options.with_mysql:
-
-            _mysql_dir = None
-            _mysql_include_dir = None
-            _mysql_libraries = None
-
-            _mysql_lib_name = None
-            if _is_os_windows:
-                _mysql_lib_name = "libmysql" if self.options["mysql-connector-c"].shared else "mysqlclient"
-            else:
-                _mysql_lib_name = "libmysqlclient"
-
-            _mysql_lib_ext = _shared_lib_ext if self.options["mysql-connector-c"].shared else _static_lib_ext
-
-            if self._set_mysql_paths():
-                _mysql_include_dir = self.deps_cpp_info["mysql-connector-c"].include_paths[0]
-                _mysql_libraries = os.path.join(
-                    self.deps_cpp_info["mysql-connector-c"].lib_paths[0],
-                    ("%s.%s" % (_mysql_lib_name, _mysql_lib_ext))
-                )
-            else:
-                if self.options.mysql_dir:
-                    _mysql_dir = self.options.mysql_dir
-
-                if self.options.mysql_include_dir:
-                    _mysql_include_dir = self.options.mysql_include_dir
-
-                if self.options.mysql_libraries:
-                    _mysql_libraries = self.options.mysql_libraries
-
-            if _mysql_dir:
-                cmake.definitions["MYSQL_DIR"] = _mysql_dir
-
-            if _mysql_include_dir:
-                cmake.definitions["MYSQL_INCLUDE_DIR"] = _mysql_include_dir
-
-            if _mysql_libraries:
-                cmake.definitions["MYSQL_LIBRARIES"] = _mysql_libraries
-
-            if self.options.soci_mysql_test_connstr:
-                cmake.definitions["SOCI_MYSQL_TEST_CONNSTR"] = \
-                    self.options.soci_mysql_test_connstr
-
         cmake.definitions["WITH_ODBC"] = self.options.with_odbc
-        if self.options.with_firebird:
-            if self._requires_odbc():
-                cmake.definitions["ODBC_INCLUDE_DIR"] = \
-                    self.options.odbc_include_dir
-                cmake.definitions["ODBC_LIBRARIES"] = \
-                    self.options.odbc_libraries
-
-            if self.options.soci_odbc_test_access_connstr:
-                cmake.definitions["SOCI_ODBC_TEST_ACCESS_CONNSTR"] = \
-                    self.options.soci_odbc_test_access_connstr
-
-            if self.options.soci_odbc_test_mysql_connstr:
-                cmake.definitions["SOCI_ODBC_TEST_MYSQL_CONNSTR"] = \
-                    self.options.soci_odbc_test_mysql_connstr
-
-            if self.options.soci_odbc_test_postgresql_connstr:
-                cmake.definitions["SOCI_ODBC_TEST_POSTGRESQL_CONNSTR"] = \
-                    self.options.soci_odbc_test_postgresql_connstr
-
         cmake.definitions["WITH_ORACLE"] = self.options.with_oracle
-        if self.options.with_oracle:
-            if self._requires_oracle():
-                cmake.definitions["ORACLE_INCLUDE_DIR"] = \
-                    self.options.oracle_include_dir
-                cmake.definitions["ORACLE_LIBRARIES"] = \
-                    self.options.oracle_libraries
-
-            if self.options.soci_oracle_test_connstr:
-                cmake.definitions["SOCI_ORACLE_TEST_CONNSTR"] = \
-                    self.options.soci_oracle_test_connstr
-
         cmake.definitions["WITH_POSTGRESQL"] = self.options.with_postgresql
-        if self.options.with_postgresql:
-
-            _postgresql_include_dir = None
-            _postgresql_libraries = None
-            _postgresql_lib_ext = _shared_lib_ext if self.options["libpq"].shared else _static_lib_ext
-            if self._set_postgresql_paths():
-                _postgresql_include_dir = self.deps_cpp_info["libpq"].include_paths[0]
-                _postgresql_libraries = os.path.join(
-                    self.deps_cpp_info["libpq"].lib_paths[0], ("libpq.%s" % _postgresql_lib_ext))
-
-            else:
-                _postgresql_include_dir = self.options.postgresql_include_dir
-                _postgresql_libraries = self.options.postgresql_libraries
-
-            if _postgresql_include_dir:
-                cmake.definitions["POSTGRESQL_INCLUDE_DIR"] = _postgresql_include_dir
-
-            if _postgresql_libraries:
-                cmake.definitions["POSTGRESQL_LIBRARIES"] = _postgresql_libraries
-
-            if self.options.soci_postgresql_test_connstr:
-                cmake.definitions["SOCI_POSTGRESQL_TEST_CONNSTR"] = \
-                    self.options.soci_postgresql_test_connstr
-
-            if self.options.soci_postgresql_nosinglerowmode:
-                cmake.definitions["SOCI_POSTGRESQL_NOSINGLEROWMODE"] = True
-
+        cmake.definitions["SOCI_POSTGRESQL_NOSINGLEROWMODE"] = \
+                self.options.soci_postgresql_nosinglerowmode
         cmake.definitions["WITH_SQLITE3"] = self.options.with_sqlite3
-        if self.options.with_sqlite3:
-            if self._requires_sqlite3():
-                cmake.definitions["SQLITE_INCLUDE_DIR"] = \
-                    self.options.sqlite_include_dir
-                cmake.definitions["SQLITE_LIBRARIES"] = \
-                    self.options.sqlite_libraries
-
-            if self.options.soci_sqlite_test_connstr:
-                cmake.definitions["SOCI_SQLITE3_TEST_CONNSTR"] = \
-                    self.options.soci_sqlite_test_connstr
-
-        # Fixes: https://github.com/SOCI/soci/pull/712
-        # Unused CMake variables DB2_LIBRARIES, ORACLE_LIBRARIES, POSTGRESQL_LIBRARY,
-        # SQLITE3_LIBRARY.
-        if "DB2_LIBRARIES" in cmake.definitions:
-            cmake.definitions["DB2_LIBRARY"] = cmake.definitions["DB2_LIBRARIES"]
-
-        if "ORACLE_LIBRARIES" in cmake.definitions:
-            cmake.definitions["ORACLE_LIBRARY"] = cmake.definitions["ORACLE_LIBRARIES"]
-
-        if "POSTGRESQL_LIBRARIES" in cmake.definitions:
-            cmake.definitions["POSTGRESQL_LIBRARY"] = cmake.definitions["POSTGRESQL_LIBRARIES"]
-
-        if "SQLITE3_LIBRARIES" in cmake.definitions:
-            cmake.definitions["SQLITE3_LIBRARY"] = cmake.definitions["SQLITE3_LIBRARIES"]
 
         cmake.configure(build_folder=self._build_subfolder)
         return cmake

